autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -244,18 +284,95 @@ def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str =
+            if "embedding_function" in self._retrieve_config:
+                self._db_config["embedding_function"] = self._embedding_function
+            self._vector_db = VectorDBFactory.create_vector_db(db_type=self._vector_db, **self._db_config)
         self.register_reply(Agent, RetrieveUserProxyAgent._generate_retrieve_user_reply, position=2)
 
+    def _init_db(self):
+        if not self._vector_db:
+            return
+
+        IS_TO_CHUNK = False  # whether to chunk the raw files
+        if self._new_docs:
+            IS_TO_CHUNK = True
+        if not self._docs_path:
+            try:
+                self._vector_db.get_collection(self._collection_name)
+                logger.warning(f"`docs_path` is not provided. Use the existing collection `{self._collection_name}`.")
+                self._overwrite = False
+                self._get_or_create = True
+                IS_TO_CHUNK = False
+            except ValueError:
+                raise ValueError(
+                    "`docs_path` is not provided. "
+                    f"The collection `{self._collection_name}` doesn't exist either. "
+                    "Please provide `docs_path` or create the collection first."
+                )
+        elif self._get_or_create and not self._overwrite:
+            try:
+                self._vector_db.get_collection(self._collection_name)
+                logger.info(f"Use the existing collection `{self._collection_name}`.", color="green")
+            except ValueError:
+                IS_TO_CHUNK = True
+        else:
+            IS_TO_CHUNK = True
+
+        self._vector_db.active_collection = self._vector_db.create_collection(
+            self._collection_name, overwrite=self._overwrite, get_or_create=self._get_or_create
+        )
+
+        docs = None
+        if IS_TO_CHUNK:
+            if self.custom_text_split_function is not None:
+                chunks = split_files_to_chunks(
+                    get_files_from_dir(self._docs_path, self._custom_text_types, self._recursive),
+                    custom_text_split_function=self.custom_text_split_function,
+                )
+            else:
+                chunks = split_files_to_chunks(
+                    get_files_from_dir(self._docs_path, self._custom_text_types, self._recursive),
+                    self._max_tokens,
+                    self._chunk_mode,
+                    self._must_break_at_empty_line,
+                )
+            logger.info(f"Found {len(chunks)} chunks.")
+
+            if self._new_docs:
+                all_docs_ids = set(
+                    [
+                        doc["id"]
+                        for doc in self._vector_db.get_docs_by_ids(ids=None, collection_name=self._collection_name)
+                    ]
+                )
+            else:
+                all_docs_ids = set()
+
+            chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks]
```

Comment:
I started getting this error on line 365 when I pulled this commit. @thinkall 
`2024-04-09 23:38:55,157 - autogen.agentchat.contrib.retrieve_user_proxy_agent - INFO - Found 2 chunks.
Traceback (most recent call last):
  File "/mnt/c/Users/knuck/Development/autogen/./pgvector_rag.py", line 70, in <module>
    ragproxyagent.initiate_chat(
  File "/mnt/c/Users/knuck/Development/autogen/autogen/agentchat/conversable_agent.py", line 987, in initiate_chat
    msg2send = message(_chat_info["sender"], _chat_info["recipient"], kwargs)
  File "/mnt/c/Users/knuck/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 628, in message_generator
    sender.retrieve_docs(problem, n_results, search_string)
  File "/mnt/c/Users/knuck/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 552, in retrieve_docs
    self._init_db()
  File "/mnt/c/Users/knuck/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 365, in _init_db
    chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks]
  File "/mnt/c/Users/knuck/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 365, in <listcomp>
    chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks]
AttributeError: 'list' object has no attribute 'encode'`

Reply:
> I started getting this error on line 365 when I pulled this commit. @thinkall
> 
> ```
> 2024-04-09 23:38:55,157 - autogen.agentchat.contrib.retrieve_user_proxy_agent - INFO - Found 2 chunks.
> Traceback (most recent call last):
>   File "/Development/autogen/./pgvector_rag.py", line 70, in <module>
>     ragproxyagent.initiate_chat(
>   File "/Development/autogen/autogen/agentchat/conversable_agent.py", line 987, in initiate_chat
>     msg2send = message(_chat_info["sender"], _chat_info["recipient"], kwargs)
>   File "/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 628, in message_generator
>     sender.retrieve_docs(problem, n_results, search_string)
>   File "/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 552, in retrieve_docs
>     self._init_db()
>   File "/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 365, in _init_db
>     chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks]
>   File "/Development/autogen/autogen/agentchat/contrib/retrieve_user_proxy_agent.py", line 365, in <listcomp>
>     chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks]
> AttributeError: 'list' object has no attribute 'encode'
> ```

Which commit? I just tried on my side and saw no issue.

Reply:
@thinkall The chunks data structure is a Tuple[List[dict]]
`chunk_ids = [hashlib.blake2b(chunk.encode("utf-8")).hexdigest()[:HASH_LENGTH] for chunk in chunks[0]]`

chunks[0] seems to have fixed this issue.

Reply:
After further investigation. The issue is commented above.

Reply:
Thanks. Done.